include/swift/Parse/ParsedRawSyntaxRecorder.h

```diff
@@ -0,0 +1,72 @@
+//===--- ParsedRawSyntaxRecorder.h - Raw Syntax Parsing Recorder ----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the ParsedRawSyntaxRecorder, which is the interface the
+// parser is using to pass parsed syntactic elements to a SyntaxParseActions
+// receiver and get a ParsedRawSyntaxNode object back.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_PARSE_PARSEDRAWSYNTAXRECORDER_H
+#define SWIFT_PARSE_PARSEDRAWSYNTAXRECORDER_H
+
+#include "swift/Basic/LLVM.h"
+#include <memory>
+
+namespace swift {
+
+class ParsedRawSyntaxNode;
+class SyntaxParseActions;
+class SourceLoc;
+class Token;
+enum class tok;
+
+namespace syntax {
+  enum class SyntaxKind;
+  struct Trivia;
+}
+
+class ParsedRawSyntaxRecorder {
+  std::shared_ptr<SyntaxParseActions> SPActions;
+
+public:
+  explicit ParsedRawSyntaxRecorder(std::shared_ptr<SyntaxParseActions> spActions)
+    : SPActions(std::move(spActions)) {}
+
+  ParsedRawSyntaxNode recordToken(const Token &tok,
+                                  const syntax::Trivia &leadingTrivia,
+                                  const syntax::Trivia &trailingTrivia);
+
+  /// Record a missing token. \c loc can be invalid or an approximate location
```

Comment:
Nitpick: You should use `\p` to annotate function arguments.

Reply:
Done.